integration-tests/test-oncotree-dependent-tools.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+TESTING_DIRECTORY=/var/lib/jenkins/tempdir
+if [ ! -d $TESTING_DIRECTORY ] ; then
+    mkdir -p $TESTING_DIRECTORY
+fi
+TESTING_DIRECTORY_TEMP=$(mktemp -d $TESTING_DIRECTORY/pr-integration.XXXXXX)
+ROOT_WORKSPACE=`pwd`
+CMO_PIPELINES_DIRECTORY=$ROOT_WORKSPACE/cmo-pipelines
+ONCOTREE_DIRECTORY=$ROOT_WORKSPACE/oncotree
+
+ONCOTREE_JAR=$ONCOTREE_DIRECTORY/web/target/oncotree.jar
+IMPORT_SCRIPTS_DIRECTORY=$CMO_PIPELINES_DIRECTORY/import-scripts
+
+JENKINS_USER_HOME_DIRECTORY=/var/lib/jenkins
+JENKINS_PROPERTIES_DIRECTORY=$JENKINS_USER_HOME_DIRECTORY/pipelines-configuration/properties
+APPLICATION_PROPERTIES=application.properties
+
+ONCOTREE_CODE_CONVERTER_TEST_SUCCESS=0
+FAKE_ONCOTREE_VERSION_TEST_SUCCESS=0
+ONCOTREE_CODE_CONVERTER_OUTPUT_TEST_SUCCESS=0
+
+# will be automatically called when script exits
+# provided $ONCOTREE_PORT is defined, will find process number for process on that port and kill it
+function find_and_kill_oncotree_process {
+    ONCOTREE_PORT_NUMBER=$1
+    if [ ! -z $ONCOTREE_PORT_NUMBER ] ; then
+        ONCOTREE_PROCESS_NUMBER=`netstat -tanp | grep LISTEN | sed 's/\s\s\s*/\t/g' | grep -P ":$ONCOTREE_PORT_NUMBER\t" | cut -f6 | sed 's/\/.*//'`
+    fi
+    if [ ! -z $ONCOTREE_PROCESS_NUMBER ] ; then
+        kill -9 $ONCOTREE_PROCESS_NUMBER
+        if [ $? -gt 0 ] ; then
+            echo "failed to kill process, please check for running oncotree process"
+        fi
+    else
+            echo "no oncotree process found"
+    fi
+}
+trap 'find_and_kill_oncotree_process $ONCOTREE_PORT' EXIT
+
+function find_free_port {
+    CHECKED_PORT=10000
+    MAX_PORT=65535
+    EXISTING_PORT=PORTNUMBER
+    # each loop sets EXISTING_PORT to line from netstat (corresponding to CHECKED_PORT)
+    # if grep CHECKED_PORT returns nothing, EXISTING_PORT is unset
+    # CHECKED_PORT is a free port since not found in netstat output
+    while [[ ! -z $EXISTING_PORT || $CHECKED_PORT -gt $MAX_PORT ]] ; do
+        CHECKED_PORT=$(($CHECKED_PORT + 1))
+        EXISTING_PORT=`netstat -tanp | sed 's/\s\s\s*/\t/g' | grep -P ":$CHECKED_PORT\t"`
+    done
+    if [ $CHECKED_PORT -gt $MAX_PORT ] ; then
+        echo -1
+    else
+        echo $CHECKED_PORT
+    fi
+}
+
+# Copy in ONCOTREE properties and build jar
+rsync $JENKINS_PROPERTIES_DIRECTORY/oncotree/$APPLICATION_PROPERTIES $ONCOTREE_DIRECTORY/web/src/main/resources
+cd $ONCOTREE_DIRECTORY ; mvn package -Dpackaging.type=jar
+
+#start up ONCOTREE on some port on dashi-dev
+ONCOTREE_PORT=`find_free_port`
+
+TIME_BETWEEN_ONCOTREE_AVAILIBILITY_TESTS=3
+ONCOTREE_DEPLOYMENT_SUCCESS=0
+CURRENT_WAIT_TIME=0
+MAXIMUM_WAIT_TIME=600
+if [ $ONCOTREE_PORT -gt 0 ] ; then
+    java -jar $ONCOTREE_JAR --server.port=$ONCOTREE_PORT &
+
+    # maximum time to wait for oncotree to deploy 2 minutes
+    # every 10 seconds check if job is still running
```

Comment:
make this "check periodically to see if job is still running" ... it is no longer 10 seconds